instant-xml/src/impls.rs

```diff
@@ -334,23 +330,93 @@ where
 
     fn deserialize(
         into: &mut Self::Accumulator,
-        field: &'static str,
+        _: &'static str,
         deserializer: &mut Deserializer<'_, 'xml>,
     ) -> Result<(), Error> {
-        if into.is_some() {
+        if into.inner.is_some() {
             return Err(Error::DuplicateValue);
         }
 
-        let mut value = <T::Owned as FromXml<'xml>>::Accumulator::default();
-        T::Owned::deserialize(&mut value, field, deserializer)?;
-        *into = Some(Cow::Owned(value.try_done(field)?));
+        let value = match deserializer.take_str()? {
+            Some(value) => value,
+            None => return Ok(()),
+        };
+
+        into.inner = Some(decode(value)?.into_owned().into());
+        Ok(())
+    }
+
+    type Accumulator = CowStrAccumulator<'xml, 'a>;
+    const KIND: Kind = Kind::Scalar;
+}
+
+#[derive(Default)]
+pub struct CowStrAccumulator<'xml, 'a> {
+    pub(crate) inner: Option<Cow<'a, str>>,
+    marker: PhantomData<&'xml str>,
+}
+
+impl<'xml, 'a> Accumulate<Cow<'a, str>> for CowStrAccumulator<'xml, 'a> {
+    fn try_done(self, field: &'static str) -> Result<Cow<'a, str>, Error> {
+        match self.inner {
+            Some(inner) => Ok(inner),
+            None => Err(Error::MissingValue(field)),
+        }
+    }
+}
+
+impl<'xml, 'a, T: FromXml<'xml>> FromXml<'xml> for Cow<'a, [T]>
+where
+    [T]: ToOwned<Owned = Vec<T>>,
+{
+    #[inline]
+    fn matches(id: Id<'_>, field: Option<Id<'_>>) -> bool {
+        match field {
+            Some(field) => id == field,
+            None => false,
+        }
+    }
+
+    fn deserialize(
+        into: &mut Self::Accumulator,
+        field: &'static str,
+        deserializer: &mut Deserializer<'_, 'xml>,
+    ) -> Result<(), Error> {
+        let mut value = T::Accumulator::default();
+        T::deserialize(&mut value, field, deserializer)?;
+        into.inner.push(value.try_done(field)?);
         Ok(())
     }
 
-    type Accumulator = Option<Cow<'a, T>>;
+    type Accumulator = CowSliceAccumulator<T>;
     const KIND: Kind = Kind::Scalar;
 }
 
+pub struct CowSliceAccumulator<T>
+where
+    [T]: ToOwned,
+{
+    pub(crate) inner: Vec<T>,
```

Comment:
I would expect, based on the name, that this would be a `Cow<'a, [T]>`. Maybe this should be named `VecAccumulator` if it only contains owned types?

Reply:
Oh I think I get it now. This is needed to allow deserialization of the following, right?
```rs
struct A<'a, T>{
  field: Cow<T, 'a>
}
```
While deserializing, the matching occurrences are accumulated in a Vec (using `CowSliceAccumulator`)
and then, while constructing the `A` struct, are turned back into a Cow again (by calling `try_done`)?

I guess this is ok for now. But I think as soon as we have another newtype over `Vec` to implement Accumulate<T> for the newtype, we should unify them and rename this to `VecAccumulator`.

Reply:
Ohh, fair point. I changed it to use `Vec<T>` as the `Accumulator` for `Cow<'a, [T]>` and added a generic `Accumulate` impl to allow accumulating from a `Vec<T>` into the `Cow<'a, [T]>`.